### ISSUE TITLE
L025 triggers when table alias is used in BigQuery QUALIFY clause

### ISSUE THREAD

**Issue by elyobo** [sqlfluff/sqlfluff#2646]
## Expected Behaviour

L025 "Tables should not be aliased if that alias is not used." should detect unused aliases.

## Observed Behaviour

The rule does not detect aliases used in a BigQuery `QUALIFY` window definition.

Additionally the rule error specifically talks about using the alias in the SELECT statement, but that isn't specified in the docs (and judging by https://github.com/sqlfluff/sqlfluff/issues/2613 the rule is meant to detect unused aliases elsewhere).

## Steps to Reproduce

The following example reports "Alias 'tbl2' is never used in SELECT statement.". This message is technically correct (it's not used _in the select statement_) but

* it is used, so no error should be reported
* the message does not reflect the intent of the rule (so long as its used somewhere that should be sufficient, usage in the SELECT specifically is not required)

```sql
SELECT
  tbl1.col1 AS col1
FROM
  tbl1 AS tbl1
  JOIN tbl2 AS tbl2 ON (tbl2.col2 = tbl1.col2)
WHERE TRUE
QUALIFY ROW_NUMBER() OVER (
  PARTITION BY tbl1.col1
  ORDER BY tbl2.col3
 ) = 1
```

## Dialect

BigQuery

## Version
sqlfluff
```
sqlfluff, version 0.8.2
```

dbt
```
installed version: 0.21.1
   latest version: 1.0.0

Your version of dbt is out of date! You can find instructions for upgrading here:
https://docs.getdbt.com/docs/installation

Plugins:
  - snowflake: 0.21.1
  - redshift: 0.21.1
  - bigquery: 0.21.1
  - postgres: 0.21.1
```

## Configuration

```
[tool.sqlfluff.core]
verbose = 0
nocolor = false
dialect = "bigquery"
templater = "dbt"
recurse = 0
output_line_length = 80
runaway_limit = 10
ignore_templated_areas = true
encoding = "autodetect"
# Comma separated list of file extensions to lint.
# NB: This config will only apply in the root folder.
sql_file_exts = ".sql,.sql.j2,.dml,.ddl"

# Disable some rules
# Allow aliasing in FROM/JOIN (L031)
# Allow USING statement
# Disable wildcard over specific select (L034) - it causes some unwanted behaviour, specifically
# complaining about select * in subqueries
exclude_rules = "L031,L032,L034"

[tool.sqlfluff.templater.dbt]
project_dir = "./"
profiles_dir = "./"
profile = "property_management"

[tool.sqlfluff.indentation]
indented_joins = false
indented_using_on = false
template_blocks_indent = true

[tool.sqlfluff.templater]
unwrap_wrapped_queries = true

[tool.sqlfluff.templater.jinja]
apply_dbt_builtins = true

[tool.sqlfluff.templater.jinja.macros]
# Macros provided as builtins for dbt projects
dbt_ref = "{% macro ref(model_ref) %}{{model_ref}}{% endmacro %}"
dbt_source = "{% macro source(source_name, table) %}{{source_name}}_{{table}}{% endmacro %}"
dbt_config = "{% macro config() %}{% for k in kwargs %}{% endfor %}{% endmacro %}"
dbt_var = "{% macro var(variable, default='') %}item{% endmacro %}"
dbt_is_incremental = "{% macro is_incremental() %}true{% endmacro %}"
load_macros_from_path="macros"

# Some rules can be configured directly from the config common to other rules.
[tool.sqlfluff.rules]
indent_unit = "space"
tab_space_size = 2
max_line_length = 100
comma_style = "trailing"
allow_scalar = true
single_table_references = "consistent"
unquoted_identifiers_policy = "all"

# Some rules have their own specific config.
[tool.sqlfluff.rules.L007]  # Keywords
operator_new_lines = "after"

[tool.sqlfluff.rules.L010]  # Keywords
capitalisation_policy = "upper"

[tool.sqlfluff.rules.L011]  # Aliasing
aliasing = "explicit"

[tool.sqlfluff.rules.L012]  # Aliasing
aliasing = "explicit"

[tool.sqlfluff.rules.L014]  # Unquoted identifiers
extended_capitalisation_policy = "lower"

[tool.sqlfluff.rules.L016]
ignore_comment_lines = false

[tool.sqlfluff.rules.L026]
force_enable = false

[tool.sqlfluff.rules.L028]
force_enable = false

[tool.sqlfluff.rules.L029]  # Keyword identifiers
unquoted_identifiers_policy = "aliases"

[tool.sqlfluff.rules.L030]  # Function names
capitalisation_policy = "upper"

[tool.sqlfluff.rules.L038]
select_clause_trailing_comma = "require"

[tool.sqlfluff.rules.L040]  # Null & Boolean Literals
capitalisation_policy = "upper"

[tool.sqlfluff.rules.L042]
# By default, allow subqueries in from clauses, but not join clauses.
forbid_subquery_in = "join"

[tool.sqlfluff.rules.L047]  # Consistent syntax to count all rows
prefer_count_1 = false
prefer_count_0 = false
```

**Comment by barrywhart** [comment]
I'm going to use this query to test and fix the issue:
```
SELECT *
FROM
    table1 AS tbl1
INNER JOIN tbl2 AS tbl2
WHERE TRUE
QUALIFY ROW_NUMBER() OVER (
    PARTITION BY tbl1.col1
    ORDER BY tbl2.col3
    ) = 1
```

Note that today's fix to #2613 partly addresses the issues with this query, because it updates L025 to look at the `JOIN` clause. Still need to update it to look at the `QUALIFY`, though. I think this will be a quick fix.

**Comment by elyobo** [comment]
Cool, thanks @barrywhart!
